syncd/syncd_flex_counter.h

```diff
@@ -198,6 +199,20 @@ class FlexCounter
         bool allIdsEmpty();
         bool allPluginsEmpty();
 
+        typedef void (FlexCounter::*collect_counters_handler_t)(_In_ swss::Table &countersTable);
+        typedef std::unordered_map<std::string, collect_counters_handler_t> collect_counters_handler_unordered_map_t;
+
+        void collectPortCounters(_In_ swss::Table &countersTable);
+        void collectQueueCounters(_In_ swss::Table &countersTable);
+        void collectQueueAttrs(_In_ swss::Table &countersTable);
+        void collectPriorityGroupCounters(_In_ swss::Table &countersTable);
+        void collectPriorityGroupAttrs(_In_ swss::Table &countersTable);
+        void collectRifCounters(_In_ swss::Table &countersTable);
+        void collectBufferPoolCounters(_In_ swss::Table &countersTable);
```

Comment:
Along the lines of Kamil's request to try to generalize things, would it make sense to have a generic collectCounters() method and pass in the counter type key (e.g., `void collectPortCounters(_In_ const std::string &key, _In_ swss::Table &countersTable);`, then incorporate logic in that single method to collect the appropriate counter type based on the key provided. This would eliminate a log of duplicate code, but depending on how much code can't be shared, the method could potentially become long and hard to follow. That requires a judgement call.

Reply:
This can be our ultimate goal. We can achieve it step by step. https://github.com/Azure/sonic-sairedis/pull/488#pullrequestreview-265057977